File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/model/MethodDescriptionParser.java
package eu.nyerel.panda.ijplugin.model;

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
public enum MethodDescriptionParser {

    INSTANCE;

    public Method parse(String description) {
        int lastPeriod = description.substring(0, description.lastIndexOf("(")).lastIndexOf(".");
        String className = description.substring(0, lastPeriod);
        String signature = description.substring(lastPeriod + 1);
        String methodName = signature.substring(0, signature.indexOf("("));
        lastPeriod = className.lastIndexOf(".");
        String packageName = className.substring(0, lastPeriod);
        String classNameSimple = className.substring(lastPeriod + 1);

        Method method = new Method();
        method.setClassName(className);
        method.setClassNameSimple(classNameSimple);
        method.setSignature(signature);
        method.setSignatureFull(description);
        method.setName(methodName);
        method.setPackageName(packageName);

        return method;
    }

}

File: panda-ij-plugin/src/eu/nyerel/panda/ijplugin/model/MethodDescriptionParser.kt
package eu.nyerel.panda.ijplugin.model

/**
 * @author Rastislav Papp (rastislav.papp@gmail.com)
 */
object MethodDescriptionParser {

    fun parse(description: String): Method {
        var lastPeriod = description.substring(0, description.lastIndexOf("(")).lastIndexOf(".")
        val className = description.substring(0, lastPeriod)
        val signature = description.substring(lastPeriod + 1)
        val methodName = signature.substring(0, signature.indexOf("("))
        lastPeriod = className.lastIndexOf(".")
        val packageName = className.substring(0, lastPeriod)
        val classNameSimple = className.substring(lastPeriod + 1)
        val args = parseArgs(signature)

        return Method(className, classNameSimple, packageName, signature, description,  args, methodName)
    }

    private fun parseArgs(signature: String): List<String> {
        val args = signature.substringAfter("(").substringBefore(")")
        return if (args.isNotEmpty()) {
            args.split(",")
        } else {
            emptyList()
        }
    }

}
